modules/460-log-shipper/hooks/internal/vector/destination/loki.go

```diff
@@ -112,6 +118,33 @@ func NewLoki(name string, cspec v1alpha1.ClusterLogDestinationSpec) *Loki {
 		tls.VerifyHostname = *spec.TLS.VerifyHostname
 	}
 
+	endpoint, err := url.Parse(spec.Endpoint)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse loki endpoint '%s'", spec.Endpoint)
+	}
+
+	lokiAddr, err := net.ResolveTCPAddr("tcp", endpoint.Host)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to resolve ip address for loki endpoint '%s'", spec.Endpoint)
+	}
+
+	deckhouseLokiAddr, err := net.ResolveTCPAddr("tcp", d8LokiPodAddress)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to resolve ip address for deckhouse loki pod address '%s'", d8LokiPodAddress)
+	}
+
+	if lokiAddr.IP.Equal(deckhouseLokiAddr.IP) && lokiAddr.Port == deckhouseLokiAddr.Port {
+		endpoint.Scheme = "https"
+
+		spec.Endpoint = endpoint.String()
+
+		spec.Auth.Strategy = "Bearer"
+		spec.Auth.Token = token
+
+		tls.VerifyHostname = false
+		tls.VerifyCertificate = false
+	}
```

Comment:
Let's leave a detailed explanation in the comment on why we need this block. Provide as much info as possible about migration.

Reply:
done